test.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+set -e
+
+function printUsage() {
+  echo "Usage: test.sh <repository> <username>"
+  echo "Example: test.sh compucorp/rse davialexandre"
+  echo "The script expects the following environment variables to be defined in order for it to run:"
+  echo "  GITHUB_TOKEN: a personal access token from the given user with permissions to clone and create pull requests to the given repository"
+  echo "  AWS_ACCESS_KEY: a access key with permission to read from the anonymized-db S3 bucket used by the mysql-anondb docker image"
+  echo "  AWS_SECRET_KEY: the secret key for the given AWS_ACCESS_KEY"
+}
+
+function getContainerStatus() {
+  docker inspect --format="{{if .Config.Healthcheck}}{{print .State.Health.Status}}{{end}}" "$1"
+}
+
+[ -z "$1" ] && echo "Missing repository!" && printUsage && exit 1
+[ -z "$2" ] && echo "Missing username!" && printUsage && exit 1
+[ -z "$GITHUB_TOKEN" ] && echo "Missing GITHUB_TOKEN" && printUsage && exit 1
+[ -z "$AWS_ACCESS_KEY" ] && echo "Missing AWS_ACCESS_KEY" && printUsage && exit 1
+[ -z "$AWS_SECRET_KEY" ] && echo "Missing AWS_ACCESS_KEY" && printUsage && exit 1
+
+set -x
+
+repository=$1
+github_username=$2
+
+## Clone the repository following an approach similar to what https://github.com/actions/checkout does
+workdir="$PWD/workdir/$repository"
+mkdir -p "$workdir"
+cd "$workdir"
+git init
+git remote add origin "https://$github_username:$GITHUB_TOKEN@github.com/$repository"
+git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules origin "+refs/heads/*:refs/remotes/origin/*" "+refs/tags/*:refs/tags/*"
+git checkout --progress --force -B master refs/remotes/origin/master
+
+# We need the FROM_SITE in order to start the mysql-anondb container
+from_site=$(grep -Eoh "FROM_SITE: (.+?)" "$workdir"/.github/workflows/* | cut -d ' ' -f 2)
```

Comment:
Why not to specifiy the workflow file name here : 

```
$workdir"/.github/workflows/security-updates.yaml
```

instead of looking into all the files inside the `workflows` directory .

Same with the `greps` below

Reply:
@omarabuhussein the reason for this is that the file name might not always be exactly the same. For example, some of them might have the `yml` extension instead of `yaml` and some might simply have a different name (example: https://github.com/compucorp/phabkids/blob/master/.github/workflows/security-updates-new.yaml)

Of course I could make the script handle all these cases, but since this was only for local testing I thought the current implement was good enough.

Note: if you curious why the phabkids workflow name is different, that was because of a bug on Github Actions that would not recognize updates to a workflow after I changed the file extension :man_shrugging:

Reply:
I see, then I guess it is fine to keep things as is